### ISSUE TITLE
Handling large datasets

### ISSUE THREAD

**Issue by joeroe** [nevrome/currycarbon#2]
The attached file [all_c14.txt](https://github.com/nevrome/currycarbon/files/6770687/all_c14.txt) includes 145559 dates (everything from c14bazAAR with lab IDs that currycarbon can parse).

A simple calibration with no output works very well:

```bash
$ time ./currycarbon calibrate -i all_c14.txt > /dev/null
Calibrating...
Done

real	0m0.967s
user	0m3.400s
sys	0m2.218s
```

But asking for output with `--hdrFile` or `-q` fails after maxing out my available memory (~30 GB):

```bash
$ time ./currycarbon calibrate -i all_c14.txt
 --hdrFile all_c14_cal
Calibrating...
Killed

real	1m52.308s
user	12m39.357s
sys	1m26.835s
```

**Comment by nevrome** [comment]
Alright - I solved this one, I believe. I didn't think hard enough about my parallel code and forced full evaluation of the date list before writing the results to file. Will be fixed in the next release.

**Comment by nevrome** [comment]
This should be fixed in v0.6.0
